### PR TITLE
Move (most) javascript out of jelly files

### PR DESCRIPTION
**Pull request by rahulsom**
We're limited in what we can do in jelly files.
There are tests that use HtmlUnit and it does not support enough modern JS features.

This change will allow us to use modern JS to write code, and transpile it to something that all the older browsers support and HtmlUnit does not error out.

This is built on top of https://github.com/jenkinsci/active-choices-plugin/pull/75

### Testing done

This is a pure refactor. No new functionality is added. All existing tests pass.
The code that was moved from jelly to js has a lot of dependencies.
We'll need more refactoring before we can test it independently.

```[tasklist]
### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

File: pom.xml
Comment (basil) [inline pom.xml:202]: This configuration is already provided in the parent POM if you check in a blank `.mvn_exec_node` file:

https://github.com/jenkinsci/plugin-pom/blob/a60e3494e8447aefed562deace5d222e4777413a/pom.xml#L1128-L1198

**Comment by basil** [comment]
You don't have to switch to NPM, either: the plugin parent POM has a coresponding `.mvn_exec_yarn` feature.

**Comment by rahulsom** [comment]
> You don't have to switch to NPM, either: the plugin parent POM has a corresponding .mvn_exec_yarn feature.

I don't have strong feelings one way or the other. I'll leave it at NPM for now unless @basil or @kinow feel strongly about either tool.

**Comment by basil** [comment]
Core uses Yarn, but for individual plugins it is up to the maintainer.

**Comment by kinow** [comment]
(sorry, other merges caused some conflicts here)

**Comment by rahulsom** [comment]
NP. I'll clean this up after #75 is merged.

**Comment by rahulsom** [comment]
@kinow This is now ready for review. Thanks!

File: src/main/resources/org/biouno/unochoice/stapler/unochoice/UnoChoice.es6
Comment (kinow) [inline src/main/resources/org/biouno/unochoice/stapler/unochoice/UnoChoice.es6:366]: The other changes were easier to review. This one is the only one I didn't understand straight away. Could you explain why the `false` added here?

Comment (rahulsom) [reply src/main/resources/org/biouno/unochoice/stapler/unochoice/UnoChoice.es6:366]: Sorry about that. The default value was `undefined` which was interpreted as `false`. This makes it explicit.

**Review by kinow — APPROVED** [review]

**Comment by kinow** [comment]
@rahulsom thank you for your recent contributions. I think we got all your pull requests reviewed and merged, right? Shall I cut a new release with all the current changes in `master`? Or do you have anything you'd like to include?

**Comment by rahulsom** [comment]
Thanks @kinow ! I'm working on one last PR to remove the dependency on prototype. That should get this plugin ready for the next LTS.

**Comment by kinow** [comment]
Brilliant! Thanks heaps, I have a meeting with @imoutsatsos tomorrow to chat about Jenkins and about this plug-in -- he's the person who had the idea and initial design of how this plug-in should work, as well as designer of most of the functionality that he uses at his $job. Great timing for a new release soon before summer starts over here. The next PR's are a lot easier to review now that we have a lot more (and better) tests. Bruno.

**Comment by kinow** [comment]
@eplodn possibly, this `SyntaxError` could be because of this one, I think it was the only major change in JS, and the #79 , but that one wouldn't result in a `SyntaxError`, I think.

**Comment by rahulsom** [comment]
JENKINS-71706 claims to be on 2.6.5. This change made it into 2.7.

**Pull request merged**
